packages/graphql/test/namepolicies.test.tsx

```diff
@@ -173,13 +192,72 @@ it("appends _ to reserved words in argument names", () => {
     type User {
       id: ID
     }
-    
+
     type Query {
       user(fragment_: String, mutation_: Boolean): User
     }
   `);
 });
 
+it("appends _ to lowercase SDL keywords when used as field names", () => {
+  const result = toGraphQLText(
+    <gql.ObjectTypeDefinition name="Example">
+      <gql.FieldDefinition name="enum" type={builtInScalars.String} />
+      <gql.FieldDefinition name="type" type={builtInScalars.String} />
+      <gql.FieldDefinition name="input" type={builtInScalars.String} />
+      <gql.FieldDefinition name="interface" type={builtInScalars.String} />
+      <gql.FieldDefinition name="union" type={builtInScalars.String} />
+      <gql.FieldDefinition name="scalar" type={builtInScalars.String} />
+    </gql.ObjectTypeDefinition>,
+  );
+  expect(result).toRenderTo(d`
+    type Example {
+      enum_: String
+      type_: String
+      input_: String
+      interface_: String
+      union_: String
+      scalar_: String
+    }
+  `);
+});
+
+it("appends _ to lowercase SDL keywords when used as argument names", () => {
+  const result = toGraphQLText(
+    <gql.ObjectTypeDefinition name="Query">
+      <gql.FieldDefinition
+        name="search"
+        type={builtInScalars.String}
+        args={
+          <>
+            <gql.InputValueDefinition
+              name="enum"
+              type={builtInScalars.String}
+            />
+            <gql.InputValueDefinition
+              name="type"
+              type={builtInScalars.String}
+            />
+            <gql.InputValueDefinition
+              name="input"
+              type={builtInScalars.String}
+            />
+            <gql.InputValueDefinition
+              name="interface"
+              type={builtInScalars.String}
+            />
+          </>
+        }
+      />
+    </gql.ObjectTypeDefinition>,
+  );
+  expect(result).toRenderTo(d`
+    type Query {
+      search(enum_: String, type_: String, input_: String, interface_: String): String
+    }
+  `);
+});
+
 it("appends _ to all built-in scalar names (Int, Float, String, Boolean), except ID", () => {
```

Comment:
If we follow https://github.com/pinterest/alloy/pull/13/commits/0ff0ca0d49307ed9b95367e445b98044d3d59a56#r2558545818, then all of these test cases are invalid.